omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,208 @@
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		handle(req);
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		assertEquals(LOCATION,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation().getUuid());
+		assertEquals(SOURCE,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getSource().getUuid());
+		assertTrue(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertTrue(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+		boolean automaticGenerationEnabled = false;
+		boolean manualEntryEnabled = false;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", NEW_LOCATION);
+		autoGenerationOption.add("source", NEW_SOURCE);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertEquals(NEW_LOCATION,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation().getUuid());
+		assertEquals(NEW_SOURCE,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getSource().getUuid());
+		assertFalse(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertFalse(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+	}
+	
+	@Test(expected = org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException.class)
+	public void shouldNotEditAnAutoGenerationOptionProvidedWithAllNullParams() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOptionByAnyNumberOFArguments() throws Exception {
+		assertTrue(
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isAutomaticGenerationEnabled());
+		assertTrue(identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).isManualEntryEnabled());
+		boolean automaticGenerationEnabled = false;
```

Comment:
What is the use of these two local variables? Can't you just pass in the actual values?